src/main/kotlin/net/atos/zac/app/zaak/ZaakRestService.kt

```diff
@@ -316,19 +315,18 @@ class ZaakRestService @Inject constructor(
         val zaak = zrcClientService.readZaak(zaakUUID)
         assertPolicy(policyService.readZaakRechten(zaak).wijzigen)
 
-        val oldFinalDate = zaak.uiterlijkeEinddatumAfdoening
-        val newFinalDate = restZaakEditMetRedenGegevens.zaak.uiterlijkeEinddatumAfdoening
-        val durationDays = if (newFinalDate != null) ChronoUnit.DAYS.between(oldFinalDate, newFinalDate) else null
-
         val updatedZaak = zrcClientService.patchZaak(
             zaakUUID,
             restZaakConverter.convertToPatch(restZaakEditMetRedenGegevens.zaak),
             restZaakEditMetRedenGegevens.reden
         )
 
-        if (durationDays != null) {
-            if (verlengOpenTaken(zaakUUID, durationDays, AANVULLENDE_INFORMATIE_TASK_NAME) > 0) {
-                eventingService.send(ScreenEventType.ZAAK_TAKEN.updated(updatedZaak))
+        val oldFinalDate = zaak.uiterlijkeEinddatumAfdoening
+        restZaakEditMetRedenGegevens.zaak.uiterlijkeEinddatumAfdoening?.let { newFinalDate ->
+            if (newFinalDate.isBefore(oldFinalDate)) {
+                if (adjustFinalDateForOpenTasks(zaakUUID, newFinalDate) > 0) {
```

Comment:
```suggestion
            if (newFinalDate.isBefore(oldFinalDate) && adjustFinalDateForOpenTasks(zaakUUID, newFinalDate) > 0) {
```

? Bit easier to read.